### PR TITLE
ImmutableList::every implementation

### PR DESCRIPTION
**Pull request by Protryon**
Question: is `Nil:all` returning `true` the expected behavior?

Addresses #67.

### CODE REVIEW

**Comment by bakkot** [comment]
> Question: is `Nil:all` returning `true` the expected behavior?

[Yes.](https://en.wikipedia.org/wiki/Vacuous_truth)

File: src/main/java/com/shapesecurity/functional/data/NonEmptyImmutableList.java [hunk 1]
```diff
@@ -358,6 +358,18 @@ public boolean contains(@Nonnull T a) {
         return from(result);
     }
 
+    @Override
+    public boolean all(@Nonnull F<T, Boolean> f) {
```
Comment (bakkot) [inline src/main/java/com/shapesecurity/functional/data/NonEmptyImmutableList.java]: I would prefer this be located next to `exists`, and ideally the two should be written the same way.

Comment (bakkot) [reply src/main/java/com/shapesecurity/functional/data/NonEmptyImmutableList.java]: Also, if we're going to have both `exists` and `any`, we should have `every` in addition to `all`.

Or we could just have `exists` and `every`.

Comment (bakkot) [reply src/main/java/com/shapesecurity/functional/data/NonEmptyImmutableList.java]: I think I prefer just having `exists` and `every`. Aliases seem bad.

Comment (Protryon) [reply src/main/java/com/shapesecurity/functional/data/NonEmptyImmutableList.java]: Renamed to `every`, `all` deleted.

File: src/main/java/com/shapesecurity/functional/data/NonEmptyImmutableList.java [hunk 2]
```diff
@@ -308,6 +308,18 @@ public boolean exists(@Nonnull F<T, Boolean> f) {
         }
     }
 
+    @Override
+    public boolean every(@Nonnull F<T, Boolean> f) {
+        ImmutableList<T> list = this;
+        while (list instanceof NonEmptyImmutableList) {
```
Comment (bakkot) [inline src/main/java/com/shapesecurity/functional/data/NonEmptyImmutableList.java]: Can we change the implementation to match `exists`? There's no real reason for them to be different.

**Pull request merged**